### ISSUE TITLE
Overriding https in rootUrl for local development

### ISSUE THREAD

**Issue by TJTorola** [auth0/auth0.js#930]
### Description

My team has implemented an auth0 webAuth flow for our product. All is generally well except for our local development story. The problem is that I don't want to require every member of our team to set up a Auth0 tenant along with the appropriate ngrok tunnels to do local development against local instances of our application. So our goal was to create a mock service that we could use in place of that. The problem here is that our localhost instances are not behind ssl, and the Auth0.js client forces https for the webAuth right [here](https://github.com/auth0/auth0.js/blob/b6e167d12956f3c56d179cfcae3e3f9bda0ff7e9/src/authentication/index.js#L74) if I'm not mistaken. I understand the value of using https, but for local development this is kind of putting us up against a wall where we have to do a pretty ugly hack or move _all_ of our local development services behind https and figure out how to manage certificates and all the other complexity that comes with that.

It would be _awesome_ if there was some simple override that we could use to get around this. Is that at all feasible? I'd be happy to put up a PR for it if that sounds like something that would be acceptable.

### Prerequisites

- [x] I have checked the [documentation for this library](https://auth0.github.io/auth0.js/index.html).

- [x] I have checked the [Auth0 Community](https://community.auth0.com/) for related posts.

- [x] I have checked for related or duplicate [Issues](https://github.com/auth0/auth0.js/issues) and [PRs](https://github.com/auth0/auth0.js/pulls).

- [x] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md).

- [x] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Environment

Please provide the following:

- Version of Auth0.js used: ^9.10.0
- Browser and OS version(s) affected: All
- Additional libraries or browser extensions being used that might affect behavior: NA

### Reproduction

- Input domain field for Auth0.js config, see that it defaults to `https`.

**Comment by TJTorola** [comment]
A little extra context here, I went ahead and forked the library here https://github.com/Hustle/auth0.js which worked well enough for a fork and was able to add a `_unsafeUseHttp` flag to support a mock service for local dev. But then upon moving to our react-native client I noticed that this is already basically supported by just passing in a domain that includes a protocol here https://github.com/auth0/react-native-auth0/blob/master/networking/index.js#L19.

Would it be reasonable to add the same behavior to this client instead of the route that I went with my fork? Again I would be happy to put up a PR in either direction.
